src/query/somepath.go

```diff
@@ -4,60 +4,88 @@ import "fmt"
 
 import "github.com/thought-machine/please/src/core"
 
-// SomePath finds and returns a path between two targets.
+// SomePath finds and returns a path between two targets, or between one and a set of targets.
 // Useful for a "why on earth do I depend on this thing" type query.
-func SomePath(graph *core.BuildGraph, label1 core.BuildLabel, label2 core.BuildLabel) {
-	// Awkwardly either target can be :all. This is an extremely useful idiom though so despite
-	// trickiness is worth supporting.
-	// Of course this calculation is also quadratic but it's not very obvious how to avoid that.
-	if label1.IsAllTargets() {
-		for _, target := range graph.PackageOrDie(label1).AllTargets() {
-			if querySomePath1(graph, target, label2, false) {
+func SomePath(graph *core.BuildGraph, from, to []core.BuildLabel) {
+	from = expandAllTargets(graph, from)
+	to = expandAllTargets(graph, to)
+	s := somepath{
+		memo: map[core.BuildLabel]map[core.BuildLabel]struct{}{},
+	}
+	for _, l1 := range expandAllTargets(graph, from) {
+		for _, l2 := range expandAllTargets(graph, to) {
+			if path := s.SomePath(graph.TargetOrDie(l1), graph.TargetOrDie(l2)); len(path) != 0 {
+				fmt.Println("Found path:")
+				for _, l := range filterPath(path) {
+					fmt.Printf("  %s\n", l)
+				}
 				return
 			}
 		}
-		fmt.Printf("Couldn't find any dependency path between %s and %s\n", label1, label2)
-	} else {
-		querySomePath1(graph, graph.TargetOrDie(label1), label2, true)
 	}
+	log.Fatalf("Couldn't find any dependency path between those targets")
```

Comment:
Is this what we want to do here? I guess exit 1 can be useful so you can use this in bash if statements but it feels wrong to log that as a fatal error.  Maybe `SomePath(...)` should return an exit code.

Reply:
Still exiting with 1 but now just printing instead of logging (so I can leave most of the tests as they were).

That is a change but I think it's more useful to be able to tell the difference.